### PR TITLE
Move OTA max data buffer number to project configurations.

### PR DESCRIPTION
**Pull request by ActoryOu**
Move OTA max data buffer number to project configurations.

Description
-----------

As #96, #97, #98 describe, there are compile error messages with OTA demo enabled.
- <ROOT>/components/FreeRTOS-Libraries-Integration-Tests/FreeRTOS-Libraries-Integration-Tests/src/ota_pal_test.cmake not found, and that file is removed at the latest release.
- Missing `otaconfigMAX_NUM_OTA_DATA_BUFFERS` definition.

Test Steps
-----------

1. idf.py menuconfig (Featured FreeRTOS IoT Integration -> Enable OTA demo -> check)
1. idf.py build

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [NA] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

#96, #97, #98

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo.c [hunk 1]
```diff
@@ -898,7 +897,7 @@ static uint16_t getFreeOTABuffers( void )
 
     if( xSemaphoreTake( bufferSemaphore, portMAX_DELAY ) == pdTRUE )
     {
-        for( ulIndex = 0; ulIndex < MAX_NUM_OF_OTA_DATA_BUFFERS; ulIndex++ )
+        for( ulIndex = 0; ulIndex < otaconfigMAX_NUM_OTA_DATA_BUFFERS; ulIndex++ )
```
Comment (chinglee-iot) [inline main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo.c:900]: Suggest we use the name `otademoconfigMAX_NUM_OTA_DATA_BUFFERS`
```suggestion
        for( ulIndex = 0; ulIndex < otademoconfigMAX_NUM_OTA_DATA_BUFFERS; ulIndex++ )
```

File: components/FreeRTOS-Libraries-Integration-Tests/CMakeLists.txt [hunk 2]
```diff
@@ -8,7 +8,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/FreeRTOS-Libraries-Integration-Tests/src/trans
 include(${CMAKE_CURRENT_LIST_DIR}/FreeRTOS-Libraries-Integration-Tests/src/mqtt_test.cmake)
 
 # This gives OTA_PAL_TEST_SOURCES, and OTA_PAL_TEST_INCLUDE_DIRS
-include(${CMAKE_CURRENT_LIST_DIR}/FreeRTOS-Libraries-Integration-Tests/src/ota_pal_test.cmake)
+# include(${CMAKE_CURRENT_LIST_DIR}/FreeRTOS-Libraries-Integration-Tests/src/ota_pal_test.cmake)
```
Comment (chinglee-iot) [inline components/FreeRTOS-Libraries-Integration-Tests/CMakeLists.txt:11]:
```suggestion
```

File: main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo_config.h [hunk 3]
```diff
@@ -89,6 +89,11 @@
  */
 #define otademoconfigDEMO_TASK_STACK_SIZE     ( CONFIG_GRI_OTA_DEMO_DEMO_TASK_STACK_SIZE )
 
+/**
+ * @brief The number of OTA data buffer.
+ */
+#define otaconfigMAX_NUM_OTA_DATA_BUFFERS     ( CONFIG_GRI_OTA_MAX_NUM_DATA_BUFFERS )
```
Comment (chinglee-iot) [inline main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo_config.h:95]:
```suggestion
#define otademoconfigMAX_NUM_OTA_DATA_BUFFERS     ( CONFIG_GRI_OTA_MAX_NUM_DATA_BUFFERS )
```

Comment (ActoryOu) [reply main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo_config.h]: Updated

File: components/FreeRTOS-Libraries-Integration-Tests/CMakeLists.txt (hunk 2 above)
Comment (ActoryOu) [reply components/FreeRTOS-Libraries-Integration-Tests/CMakeLists.txt]: Updated

File: main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo.c (hunk 1 above)
Comment (ActoryOu) [reply main/demo_tasks/ota_over_mqtt_demo/ota_over_mqtt_demo.c]: Updated

**Review by chinglee-iot — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Pull request merged**